berkeley-mobile/StudyPact/StudyPact.swift

```diff
@@ -33,139 +33,108 @@ class StudyPact {
+                self.email = user.profile.email
+                self.saveCryptoHash(cryptoHash: cryptohash)
+                completion(true)
+            default:
                 completion(false)
-                return
             }
-            self.cryptoHash = cryptoHash
-            self.email = user.profile.email
-            self.saveCryptoHash(cryptoHash: cryptoHash)
-            completion(true)
         }
     }
-    
-    public func getGroups(completion: @escaping ([StudyGroup]) -> Void) {
-        guard let cryptoHash = self.cryptoHash,
+
+    // MARK: AuthenticateUser
+
+    public func authenticateUser(completion: @escaping (Bool) -> Void) {
+        guard let cryptohash = self.cryptoHash,
               let email = self.email,
-              let urlString = API_URLS["GET_GROUPS"],
-              let url = URL(string: urlString) else {
-            completion([])
+              let url = EndpointKey.authenticateUser.url else {
+            completion(false)
             return
         }
-        let params = ["secret_token": cryptoHash, "user_email": email]
-        postRequest(url: url, params: params) { data in
-            guard let data = data,
-                  let json = try JSONSerialization.jsonObject(with: data, options: []) as? Array<Dictionary<String, AnyObject>> else {
-                completion([])
-                return
-            }
-            var groups: [StudyGroup] = []
-            for group in json {
-                guard let className = group["class_name"] as? String,
-                      let users = group["users"] as? Array<AnyObject>,
-                      let pending = group["pending"] as? Bool else {
-                    continue
-                }
-                var studyGroupMembers: [StudyGroupMember] = []
-                for user in users {
-                    guard let userDict = user as? Dictionary<String, String>,
-                          let name = userDict["name"],
-                          let email = userDict["email"] else {
-                        continue
-                    }
-                    let profileUrlString = userDict["profile_picture"]
-                    let member = StudyGroupMember(profilePictureURL: URL(string: profileUrlString ?? ""),
-                                                  name: name, email: email,
-                                                  phoneNumber: userDict["phone"],
-                                                  facebookUsername: userDict["facebook"])
-                    studyGroupMembers.append(member)
-                }
-                let group = StudyGroup(className: className, groupMembers: studyGroupMembers, pending: pending)
-                groups.append(group)
+        let params = AuthenticateUserParams(email: email, cryptohash: cryptohash)
+        NetworkManager.shared.get(url: url, params: params, asType: AuthenticateUserDocument.self) { response in
+            switch response {
```

Comment:
any particular reason this is a get? not too knowledgeable on the difference but a quick google says that it's better to send sensitive info as a post